enclave-runtime/src/tls_ra/tls_ra_server.rs

```diff
@@ -41,12 +42,31 @@ use std::{
 	sync::Arc,
 };
 
+#[derive(Clone, Eq, PartialEq, Debug)]
+enum ProvisioningPayload {
+	Everything,
+	ShieldingKeyOnly,
+	Nothing,
+}
+
+impl From<WorkerMode> for ProvisioningPayload {
+	fn from(m: WorkerMode) -> Self {
+		match m {
+			WorkerMode::Unknown => ProvisioningPayload::Nothing,
+			WorkerMode::OffChainWorker | WorkerMode::Oracle =>
+				ProvisioningPayload::ShieldingKeyOnly,
+			WorkerMode::Sidechain => ProvisioningPayload::Everything,
+		}
+	}
+}
```

Comment:
I like it, I just wonder if we should get rid of the `WorkerMode::Unknown`, what was your thinking behind that mode?

Reply:
I think I can remove it, yes. My initial thought was to have a mode that does not need any additional cargo features. But since that is no real use case at the moment, it probably only adds complexity and confusion without any benefit 👍

Reply:
I remember now why I put it there: When calling `cargo clippy`, we don't set any of the features, i.e. neither `OffchainWorker` nor `Sidechain` - so the code does not compile for `clippy`. I see 3 solutions:
1. Call `clippy` with the appropriate features. Downside: Makes local calling of clippy more tedious
2. Introduce an `Unknown` mode, which is set, when no additional features are given
3. If none of the compiler features are set, default to either one of the options

Reply:
The 3rd option is what we have now - sidechain is active, if none of the other features are set.